lib/src/app.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:flutter/material.dart';
+
+import 'auth.dart';
+import 'components/navigator.dart';
+import 'routing/delegate.dart';
+import 'routing/parsed_route.dart';
+import 'routing/parser.dart';
+import 'routing/route_state.dart';
+
+/// Main entry point of AmiApp
+class AmiApp extends StatefulWidget {
+  const AmiApp({Key? key}) : super(key: key);
+
+  @override
+  State<AmiApp> createState() => _AmiAppState();
+}
+
+/// App state that holds states for authentication and navigation
+class _AmiAppState extends State<AmiApp> {
+  final _auth = AmiAppAuth();
+  final _navigatorKey = GlobalKey<NavigatorState>();
+
+  late final RouteState _routeState;
+  late final SimpleRouterDelegate _routerDelegate;
+  late final TemplateRouteParser _routeParser;
+
+  @override
+  void initState() {
+    /// Configure the parser with all of the app's allowed path templates.
+    _routeParser = TemplateRouteParser(
+      allowedPaths: [
+        '/signin',
+        '/settings',
+        '/home',
+        '/logs',
+        '/events/custom',
+        '/events/device',
+        '/events/profile',
+      ],
+      guard: _guard,
+      initialRoute: '/signin',
+    );
+
+    _routeState = RouteState(_routeParser);
+
+    _routerDelegate = SimpleRouterDelegate(
+      routeState: _routeState,
+      navigatorKey: _navigatorKey,
+      builder: (context) => AmiAppNavigator(
+        navigatorKey: _navigatorKey,
+      ),
+    );
+
+    // Listen for user login state and display the sign in screen when logged out.
+    _auth.addListener(_handleAuthStateChanged);
+
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const pageTransitionsTheme = PageTransitionsTheme(
+      builders: {
+        TargetPlatform.android: FadeUpwardsPageTransitionsBuilder(),
+        TargetPlatform.iOS: CupertinoPageTransitionsBuilder(),
+        TargetPlatform.linux: FadeUpwardsPageTransitionsBuilder(),
+        TargetPlatform.macOS: CupertinoPageTransitionsBuilder(),
+        TargetPlatform.windows: FadeUpwardsPageTransitionsBuilder(),
+      },
+    );
+    const themeExtensions = <ThemeExtension<dynamic>>[];
```

Comment:
hmm, why is this empty array needed? it remains empty afterwards aswell?

Reply:
Yes we might need it. I do plan to add some attributes to it later, if not needed, I'll remove this in future PRs.